### PR TITLE
Checkpoint 1 Review

### PR DESCRIPTION
**Pull request by sundayguru**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,71 @@
-# room-allocation
\ No newline at end of file
+Room Allocation
+===============
+
+One stop solution for room allocation management
+
+### Badges
```
Comment (andela-anandaa) [inline README.md]: Move the badges to below the README title.

File: README.md [hunk 2]
```diff
@@ -1 +1,71 @@
...
+- Display room details with list of members
+- Save state to sqlite database
+- Load state from sqlite database
+
+### Dependecies
+
+- Docopt `` pip install docopt ``
+- Python 2.7.X and above
+
+### Modules used
```
Comment (andela-anandaa) [inline README.md]: **README:** What's the difference between _dependencies_ and _modules used_?

File: README.md [hunk 3]
```diff
@@ -1 +1,71 @@
-# room-allocation
\ No newline at end of file
+Room Allocation
+===============
+
+One stop solution for room allocation management
```
Comment (andela-anandaa) [inline README.md]: Give more context.

File: src/amity.py [hunk 4]
```diff
@@ -0,0 +1,623 @@
...
+			
+		for row in rooms:
+			if row['type'] == 'OFFICE':
+				room = Office(str(row['name']))
+			else:
+				room = LivingSpace(str(row['name']))
+
+			if int(row['allocated']) > 0:
+				db.table_name = 'person'
+				people = db.execute('SELECT * FROM person WHERE assigned_room LIKE "%'+ room.name +'%"')
```
Comment (andela-anandaa) [inline src/amity.py]: **Code Quality:** use of `.format()` is a cleaner way of concatenating strings that using `+`.

File: src/amity.py [hunk 5]
```diff
@@ -0,0 +1,623 @@
...
+
+		person.living_space = True if room_type == 'LIVINGSPACE' else False
+		if not self.allocate(person, room_name):
+			if room:
+				room.allocate(person)
+				Util.print_line(person.name() + ' has been moved back to ' + room.name)
+
+		self.save_state_to_pickle()
+
+	def remove_person(self, args):
```
Comment (andela-anandaa) [inline src/amity.py:435]: **Code Quality:** ensure all methods/functions have _docStrings_.

File: src/amity.py [hunk 6]
```diff
@@ -0,0 +1,623 @@
...
+from fellow import Fellow
+from fileman import FileMan
+from livingspace import LivingSpace
+from migration import Migration
+from office import Office
+from staff import Staff
+from util import Util
+
+
+class Amity(FileMan):
```
Comment (andela-anandaa) [inline src/amity.py]: **Code Quality:** there is a better way for dependency injection than using  _inheritance_. `Fileman` should be a _util_.

File: src/amity.py [hunk 7]
```diff
@@ -0,0 +1,623 @@
+from random import randint
```
Comment (andela-anandaa) [inline src/amity.py:1]: **Code Quality:** Good: separating the categories of _imports_.

File: tests/test_amity.py [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+		amity = Amity('create_room')
+		args = {
+		'<room_name>':['testRoom 1','testRoom 2','testRoom 3','testRoom 1'],
+		'<room_type>':['living','office','living'	],
+		}
+		amity.rooms = []  # make sure the rooms list is empty
+		amity.run_command(args)
+		return amity
+
+	def test_0_amity_init_sets_command(self):
```
Comment (andela-anandaa) [inline tests/test_amity.py]: **Testing:** There is no need numbering the tests. What if you were to insert a new test in between others, do you go and change all the subsequent tests?

File: tests/test_room.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+		self.assertNotEqual(data,[])
+
+	def test_room_allocate_able(self):
+		room = Office('Iroko')
+		person = Staff('Anthiny','Nandaa')
+		room.allocate(person)
+		self.assertEqual(room.allocate_able(person),False)
+
+
+	"""Edge cases for init method"""
```
Comment (andela-anandaa) [inline tests/test_room.py]: **Code Quality:** docString misplaced.

**Comment by andela-anandaa** [comment]
#### README
- Your README is okay, except for a few nitpicks (in-line) above.

#### Github Commits
- Your commit messages are not consistent, please refer to the [guidelines here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices)

#### Comments
- Fairly commented code, but some methods are missing _docString_ comments.

#### Code Quality
- See in-line comments above.

#### Functionality
- Typo in the message: "Successful" -> "Successfully"
- Added 2 more functionalities: allocate and remove person; displaying list of unallocated rooms

#### Testing
- Good attempt at writing extensive tests
- Good: integrating with TravisCI and Coveralls
